tests/CLI/modules/vlan_tests.py

```diff
@@ -93,6 +95,163 @@ def test_vlan_edit(self, click):
+        router_mock.return_value = [{'hostname': 'bcr01a.ams03'}]
+        result = self.run_command(['vlan', 'create-options'])
+        print(result.output)
+        self.assert_no_fail(result)
+
+    @mock.patch('SoftLayer.CLI.formatting.no_going_back')
+    def test_vlan_list_get_pod_with_closed_announcement(self, ngb_mock):
+        ngb_mock.return_value = True
+        vlan_mock = self.set_mock('SoftLayer_Account', 'getNetworkVlans')
+        gpods_mock = self.set_mock('SoftLayer_Network_Pod', 'getAllObjects')
+        vlan_mock.return_value = {'primaryRouter': {
+            'fullyQualifiedDomainName': 'bcr01a.fra02.softlayer.com',
+            'id': 462912},
+                                  'tagReferences': ''}
+        gpods_mock.return_value = [{'backendRouterId': 462912,
+                                    'backendRouterName': 'bcr01a.fra02',
+                                    'datacenterId': 449506,
+                                    'datacenterLongName': 'Frankfurt 2',
+                                    'frontendRouterId': 335916,
+                                    'frontendRouterName': 'fcr01a.fra02',
+                                    'name': 'fra02.pod01',
+                                    'capabilities': ['SUPPORTS_SECURITY_GROUP', 'CLOSURE_ANNOUNCED']}]
+        result = self.run_command(['vlan', 'list'])
+        self.assert_no_fail(result)
+
+    @mock.patch('SoftLayer.CLI.formatting.no_going_back')
+    def test_vlan_list_get_pod_with_closed_announcement_no_closure(self, ngb_mock):
+        ngb_mock.return_value = True
+        vlan_mock = self.set_mock('SoftLayer_Account', 'getNetworkVlans')
+        gpods_mock = self.set_mock('SoftLayer_Network_Pod', 'getAllObjects')
+        vlan_mock.return_value = {
+            'primaryRouter': {'fullyQualifiedDomainName': 'bcr01a.fra02.softlayer.com',
+                              'id': 462912
+                              },
+            'tagReferences': ''}
+        gpods_mock.return_value = [{'backendRouterId': 462912,
+                                    'backendRouterName': 'bcr01a.fra02',
+                                    'datacenterId': 449506,
+                                    'datacenterLongName': 'Frankfurt 2',
+                                    'frontendRouterId': 335916,
+                                    'frontendRouterName': 'fcr01a.fra02',
+                                    'name': 'fra02.pod01',
+                                    'capabilities': ['SUPPORTS_SECURITY_GROUP']}]
+        result = self.run_command(['vlan', 'list'])
+        self.assert_no_fail(result)
+
+    @mock.patch('SoftLayer.CLI.formatting.no_going_back')
+    def test_vlan_details_no_vs_no_hard_no_trunck(self, ngb_mock):
+        ngb_mock.return_value = True
+        vlan_mock = self.set_mock('SoftLayer_Network_Vlan', 'getObject')
+        get_object = {'primaryRouter': {'fullyQualifiedDomainName': 'bcr02a.fra02.softlayer.com',
```

Comment:
For large mock object likes this, I would prefer to put them in a fixture. https://github.com/softlayer/softlayer-python/blob/master/tests/managers/block_tests.py#L433 is one example of that pattern.

Basically you would do the following:

1. `from SoftLayer.fixtures import SoftLayer_Network_Vlan`
2. in the [SoftLayer/fixtures/SoftLayer_Network_Vlan.py](https://github.com/softlayer/softlayer-python/blob/master/SoftLayer/fixtures/SoftLayer_Network_Vlan.py) file you would create a variable like `getObject_no_trunk = <what you have for get_object here>`
3. `vlan_mock.return_value = SoftLayer_Network_Vlan.getObject_no_trunk`

That helps make these tests a little easier to read. You only need to do that for long objects like this one though. The other mocks you have in these tests are fine.